std/experimental/allocator/mallocator.d

```diff
@@ -318,6 +307,38 @@ struct AlignedMallocator
         return true;
     }
 
+    /// ditto
+    version (Posix)
+    @system @nogc nothrow
+    bool alignedReallocate(ref void[] b, size_t s, uint a) shared
+    {
+        if (!s)
+        {
+            deallocate(b);
+            b = null;
+            return true;
+        }
+        auto p = alignedAllocate(s, a);
+        if (!p.ptr)
+        {
+            deallocate(b);
```

Comment:
We don't want to deallocate the original memory here - just signal failure and let the client decide.